letsencrypt/client/cli.py

```diff
@@ -0,0 +1,373 @@
+    else:
+        try:
+            # The way to get the default would be args.email = ""
+            # First try existing account
+            return account.Account.from_existing_account(config, args.email)
+        except errors.LetsEncryptClientError:
+            try:
+                # Try to make an account based on the email address
+                return account.Account.from_email(config, args.email)
+            except errors.LetsEncryptClientError:
+                return None
+
+
+def _common_run(args, config, acc, authenticator, installer):
+    if args.domains is None:
+        doms = display_ops.choose_names(installer)
+    else:
+        doms = args.domains
+
+    if not doms:
+        sys.exit("Please specify --domains, or --installer that will "
+                 "help in domain names autodiscovery")
+
+    acme = client.Client(config, acc, authenticator, installer)
+
+    # Validate the key and csr
+    client.validate_key_csr(acc.key)
+
+    if authenticator is not None:
+        if acc.regr is None:
+            try:
+                acme.register()
+            except errors.LetsEncryptClientError:
+                sys.exit("Unable to register an account with ACME server")
+
+    return acme, doms
+
+
+def run(args, config, plugins):
+    """Obtain a certificate and install."""
+    acc = _account_init(args, config)
+    if acc is None:
+        return None
+
+    if args.configurator is not None and (args.installer is not None or
+                                          args.authenticator is not None):
+        return ("Either --configurator or --authenticator/--installer"
+                "pair, but not both, is allowed")
+
+    if args.authenticator is not None or args.installer is not None:
+        installer = display_ops.pick_installer(
+            config, args.installer, plugins)
+        authenticator = display_ops.pick_authenticator(
+            config, args.authenticator, plugins)
+    else:
+        # TODO: this assume that user doesn't want to pick authenticator
+        #       and installer separately...
+        authenticator = installer = display_ops.pick_configurator(
+            config, args.configurator, plugins)
+
+    if installer is None or authenticator is None:
+        return "Configurator could not be determined"
+
+    acme, doms = _common_run(args, config, acc, authenticator, installer)
+    cert_key, cert_path, chain_path = acme.obtain_certificate(doms)
+    acme.deploy_certificate(doms, cert_key, cert_path, chain_path)
+    acme.enhance_config(doms, args.redirect)
+
+
+def auth(args, config, plugins):
+    """Obtain a certificate (no install)."""
+    acc = _account_init(args, config)
+    if acc is None:
+        return None
+
+    authenticator = display_ops.pick_authenticator(
+        config, args.authenticator, plugins)
+    if authenticator is None:
+        return "Authenticator could not be determined"
+
+    if args.installer is not None:
```

Comment:
(optional) `--installer` helps to discover domains names if `--domains` was not set (`display_ops.choose_names` in `_common_run`).

Unless you disagree I will put that comment inline.